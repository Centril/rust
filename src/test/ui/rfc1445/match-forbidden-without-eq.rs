use std::f32;

#[derive(PartialEq)]
struct Foo {
    x: u32
}

const FOO: Foo = Foo { x: 0 };

fn main() {
    let y = Foo { x: 1 };
    match y {
        FOO => { }
        //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
        _ => { }
    }

    let x = 0.0;
    match x {
        f32::INFINITY => { }
        //~^ WARN floating-point types cannot be used in patterns
        //~| WARN this was previously accepted
        //~| WARN will become a hard error in a future release
        _ => { }
    }
}
