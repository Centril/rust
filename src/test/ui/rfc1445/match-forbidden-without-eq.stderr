error: to use a constant of type `Foo` in a pattern, `Foo` must be annotated with `#[derive(PartialEq, Eq)]`
  --> $DIR/match-forbidden-without-eq.rs:13:9
   |
LL |         FOO => { }
   |         ^^^

error: floating-point types cannot be used in patterns
  --> $DIR/match-forbidden-without-eq.rs:21:9
   |
LL |         f32::INFINITY => { }
   |         ^^^^^^^^^^^^^

error: to use a constant of type `Foo` in a pattern, `Foo` must be annotated with `#[derive(PartialEq, Eq)]`
  --> $DIR/match-forbidden-without-eq.rs:13:9
   |
LL |         FOO => { }
   |         ^^^

error: floating-point types cannot be used in patterns
  --> $DIR/match-forbidden-without-eq.rs:21:9
   |
LL |         f32::INFINITY => { }
   |         ^^^^^^^^^^^^^

error: aborting due to 4 previous errors

